### DISCUSSION TITLE
[Bug] after include the apm-jdk-threading-plugin, the Java application in the garden encountered a NoClassDefFoundError error?

### DISCUSSION THREAD

**Discussion by haoyongdong** [apache/skywalking#10654] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

when I include the apm-jdk-threading-plugin, application error by NoClassDefFoundError. I have confirmed that the class "DataSourceClassLoaderFactory" can be found in the classpath.
--------------------------------------------------------------
org.springframework.web.util.NestedServletException: Handler dispatch failed; nested exception is java.lang.NoClassDefFoundError: Could not initialize class com.alibaba.nox.executor.classloader.DataSourceClassLoaderFactory
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:982)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:901)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:861)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:621)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:728)

### What you expected to happen

Could you please help to check what caused the issue? It seems that the failure to load the class was caused by the introduction of the multi-threading plugin.

### How to reproduce

...

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by haoyongdong** [comment]
The latest version of the agent that I am using. 8.14.0

**Comment by wu-sheng** [comment]
AFAIK, we never depend on this class on the agent kernel level.
Maybe something is special at your application, which we can't tell.

**Comment by haoyongdong** [comment]
before include apm-jdk-threading-plugin
![image](https://user-images.githubusercontent.com/57824480/230380664-758a50df-7480-4859-8143-af40a93fe7d6.png)

after include
![image](https://user-images.githubusercontent.com/57824480/230380855-7dcd6423-25e8-4d4a-950f-6316b846aafd.png)

Is this additional information helpful？

**Reply by wu-sheng** [reply]
It looks like you have Arthas agent. I don't know what they do, and can't be sure whether it is relative.

**Comment by haoyongdong** [comment]
I have encountered an issue where the application I am working with uses a third-party defined class loader, which became ineffective after introducing a multi-threading plugin.

**Reply by wu-sheng** [reply]
Custom classloader is widely used, and it's fine. The key is, making sure you follow general JDK class loader way. If yours is trying to isolate something(just guess), the agent wouldn't work.
And all private codes are only possible to be fixed on your own.
